packages/ui-view/src/ContextView/styles.js

```diff
@@ -0,0 +1,283 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { mirrorPlacement } from '@instructure/ui-position'
+
+import generateComponentTheme from './theme'
+
+const getPlacementStyle = (placement, theme) => {
```

Comment:
It is just a preference/suggestion, but how about writing it as a switch? I find it more readable, but this works fine too :)
```js
switch (placement) {
  case 'end-center':
  case 'end-top':
  case 'end-bottom':
  case 'center-end':
  case 'end':
    return { paddingInlineStart: theme?.arrowSize, paddingInlineEnd: '0' }
  
  case 'start-center':
  ...etc
}
```

Reply:
Yeah, I started of with a switch case, but it started to become too cumbersome so I instead started doing this way which is still hard to read, but IMO better than the huge switch case. (an argument can be made for the switch to be faster, because it doesn't have to do the `includes` part)